[18.0][MIG] mgmtsystem_nonconformity: Migration to 18.0

natuan9 opened the pull request:
**Depend on:**
- [[18.0][MIG] document_page_procedure: Migration to 18.0](https://github.com/natuan9/management-system/pull/1)
- `[[18.0][MIG] mgmtsystem_action: Migration to 18.0 ](https://github.com/OCA/management-system/pull/605)`

**Ported from v14:**
- `#525`

**Change in v18:**
 - `[remove order parameter of group_expand methods](https://github.com/odoo/odoo/pull/139294/commits/6e6bcbc6ab5cef679b17b617d821b2bb3f778af8) `

Commit:
[ADD] mgmtsystem_nonconformity: Initial commit. Import from extra-addons.
[IMP] [#9439] Added the system reference field to the model and the views.
[IMP] efficiency -> effectiveness
[FIX] group_mgmtsystem_manager XML ID
[IMP] mgmtsystem_kpi
[FIX] Group references. Reported by michaelgiesbrecht on the forum.
[ADD] Icons
[IMP] Views
[ADD] NC : Workflow, actions_ids
[ADD] Action : Review fields
[REM] NC : Effectiveness fields
[IMP] Use workflow and improve filters.
[FIX] button_cancel invisible and system_id readonly on closed state.
[FIX] Can't group by procedure_ids

Commit:
Launchpad automatic translations update.

dreispt committed:
[IMP] mgmtsystem_nonconformity: Progress

* ADD workflow reset option. ADD workflow transition from Open to Pending. CHG config tables are translatable. ADD pt translation.
* CHG process sequence logic. ADD approval records for each stage.
* CHG: finetunning user interface
* [ADD] Actions link back to NCs. [CHG] Actions UI improvements. [DEL] Action review not necessary. [ADD] NC on_change department fills automatically responsibles.
* ADD: communication & history tab for NCs
* ADD: keep history record for state changes and approvals

pt translations

Commit:
[IMP] mgmtsystem_nonconformity:: NC Department now added by an optional module + qualification fields moved from feedback to NC; actions can be linked to projects

dreispt committed:
[IMP] mgmtsystem_nonconformity: Several changes

CHG: NC analysis is required at state 'p'

CHG: added parent to group NC Causes; removed NC Category

ADD: state 'analysis', between 'draft' and 'pending'

CHG: NC flow logic organized according to the typical sequence

CHG: small fixes and finetuning

CHG: marking Apps, security review, translation pt, finetuning

Commit:
[ADD] Move wiki_quality_manual from lp:openobject-addons/6.1 to document_page_quality_manual

Commit:
[REN] wiki_* to document_page_*

Commit:
[FIX+IMP] mgmtsystem_nonconformity: license, permission and others

* licenses and dependencies
* Update hooks with addons modules from v7
* Permissions on files
* Add origins and causes data

dreispt committed:
[FIX+IMP] mgmtsystem_nonconformity: License, translation and other stuff

* CHG Renamed 'feedback' to 'claim', see explanation  in __openerp__.py
* CHG pt_pt translation
* FIX Fixing fields read-only flags based on state
* CHG Improving action <-> NC relationship (wip)
* [FIX] fixes from review comments
* [FIX] missing context; instanciation of models
* [CHG] AGPL license; set verion to 1.0
* [CHG] Selections use words instead of letters; fixed AGPL-3 reference
* [FIX] .py files licenses to AGPL

Commit:
[FIX+IMP] mgmtsystem_nonconformity: Migration scripts + fix

* NameError: global name 'context' is not defined when resetting a NC
* Skeleton for migration script
* openupgrade analysis

Commit:
[IMP+FIX] mgmtsystem_nonconformity: Migration scripts + several things

[ADD] Added migration scripts for 0.1 --> 1.0 in the modules that needed it.

mgmtsystem_nonconformity:

* o --> open c --> done state change.
* Migrating action relations to a many2many table.
* Concatening action comments in a single field

mgmtsystem_audit and mgmtsystem_review:

* o --> open c --> done state change.

[FIX] I had accidentally broken the nonconformity/action migration shortly before commit.

[FIX] Fixed "Unknown field state for domain" errors.

In 7.0, it's necessary, in a view, to have the field present in a view in order to use it
in a domain. I added an invisible "state" field in relevant views to fix the errors.

[CHG] Adjusted mgmgsystem_nonconformity to OpenERP 7.0.

Like with mgmtsystem_action, the problem comes from changes made in OpenERP's messaging system.

[FIX] Broken reference to wiki's group_id.

In OpenERP 7, document_page had a parent_id instead of wiki's group_id. I've fixed the broken references that we had left in nonconformity and audit.

[FIX] Fixed wrongful merge.

The merge of nc-extend-7.0 was incorrect. It wrongly used the 6.1 version of mgmtsystem_audit.xml and mgmtsystem_nonconformity/mgmtsystem_nonconformity.xml. Fixed this.

[CHG] Updated statusbar/buttons's layout to conform to OpenERP 7.0 style.

Instead of a group at the bottom of the form, it's now a <header> tag at the top.

Changed Actions, Audit, Nonconformity and Review.

Commit:
Launchpad automatic translations update.

Commit:
Launchpad automatic translations update.

Commit:
[CHG] Changed MediaWiki syntax in document pages to plain HTML.

document_page doesn't seem to support MediaWiki syntax anymore.

Commit:
[CHG] Changed MediaWiki syntax in document pages to plain HTML.

document_page doesn't seem to support MediaWiki syntax anymore.

Commit:
[FIX[ Migration folder names

cloutierjo committed:
[FIX] mgmtsystem_nonconformity: Bug #1162942 + bug #1162946, update mail.thread usage of notification

Commit:
[FIX] Fixed missing import in mgmtsystem_nonconformity.

"osv" was mistakenly removed from imports in nc-extend.

Commit:
[FIX] No group can create NC

Commit:
[FIX] mgmtsystem_nonconformity: Replaced deprecated osv.except_osv() with orm.except_orm()+ Fixed calls to action.case_open() which were made with explicit (cr, uid, id) args.

Now, these methods' arguments are implicitly set.

Commit:
[UPD] translation file

dreispt committed:
[IMP] mgmtsystem_nonconformity:  migrating NC form to v7 sheet style + replaced message history with chatter widget

Commit:
[FIX+IMP] mgmtsystem_nonconformity: Several things + POT

* [FIX] Remaining group_id
* [FIX] Update domain
* [UPD] POT files
* [IMP] Add company_id and its default value
* [IMP] Add company_id in views
* [ADD] mgmtsystem_*_security.xml files

Commit:
[IMP] mgmtsystem_nonconformity: 4 improvements

* Added autosubscription for Responsible user, Manager and Top manager to follow list of OpenChatter
* Autoadding the nonconformity author.
* Exception when closing NC that has non closed actions.
* Added state 'cancelled' and check for immediate actions.

Commit:
[FIX] PEP8 compliance in audit, action and nonconformity

Commit:
[FIX+IMP] mgmtsystem_nonconformity: Several things

* Improvement to NC chatter state change messages.
* Buttons that change state backwards (towards draft) are no longer highlighted.
* Prefix overwriten for Nonconformity
* Fix to integrity check error
* revert unessessary_inherit to crm.claim
* [FIX] Fix PEP8 errors after running 'flake8 . --ignore E501,F401'
* [7.0.1.0 Migration scirpts] Added mgmtsystem_action. Made mgmtsystem_nonconformity system more robust.
* [FIX] Migration scripts no longer fully dependent on openupgrade
* [FIX] Bug #1267840 add values parametter to message_auto_subscribe
* Removed redundant import from openupgrade

Commit:
Adding migrationscript for procedure for OpenUpgrade

Commit:
Adding migration script for quality manual for OpenUpgrade

gdromgoole committed:
[IMP] Documentation in the module description

dreispt committed:
Init 8.0 branch

dreispt committed:
Init 8.0 branch

llacroix committed:
Moved document_page_procedure to root folder for port

Simple fix nothing to change

Refactor move icon and splitted xml file in data/views

llacroix committed:
Moved document_page_quality_manual to root folder for port

llacroix committed:
Making installable is enough, nothing to port

llacroix committed:
[MIG] mgmtsystem_nonconformity: Migration to 8.0

Updated module as installable and removed depdencie on base_status

See

the base_status module is apparently gone so we have to work without it.
https://code.launchpad.net/~openerp-dev/openobject-addons/trunk-remove-base_status-tde/+merge/179887

Small fix to use stage_id.name instead of state

State doesn't exist anymore

Use stage_id for actions

Add support for settled, rejected instead of done/canceled... quite ugly

Separated python in each model file

Added missing methodcase_close_send_note for workflows

Updated model severity to V8

Port cause model to v8

Fix flake8 error

Fix small error in `cause` and upgraded `origin` to v8

Splitted xml views into 5 different xml files

Moved icon.png to static/description

Added tests and updated cause to v8

Added test origin and changed name of test class for cause

Removed severity redefined here

Added tests for create nonconformity

Ported fields to v8

Fix indent block

Removed ID and removed named lambdas

Simplify the name_get method

Get exceptions without importing orm

Removed size

Simplified name_get and removed id

Replace try except by self.assertRaise

Added group in forms

Reformat xml files removed tabs/space mix and 2 space indent

Fix tests

Added left and right to cause and origin

Add ondelete to parent_id

Fix tests

Improved name_get method and full coverage or origin/cause

Added workflow test

Added missing copyrights

Fix flake errors

Fix more flake errors

Some actions can only be done once actions, analysis

Fix even more flakes errors

Test reset button

Added missing fields (used in demo data)

Test state nonconformity name

Remove states as we are using stages on actions...

It would make more sense to add a property to stages that mark some
stages as ending stages instead of checking by names.

Use stage is ending and is starting instead of by name

Test analysis instead of evaluation

Fix flake8 errors

Added base security

Security rule in wrong project

Commit:
[IMP] Add README.rst and data directory

Commit:
[FIX] No reference to document_page_group_procedure

llacroix committed:
Moved icon and reformatted manifest

llacroix committed:
Review changes from PR#35

Removed certificate from __openerp__

Remove explicit view ids in action window

sysadminmatmoz committed:
Slovene translations added

Commit:
[IMP+FIX] mgmtsystem_nonconformity: Cleanup + tests + others

* Move datas to their own folders
* Cleanup Nonconformity
* Use track and mail message subtypes instead of overriding functions
* Remove unused functions
* Cleanup and full use of v8 and track features
* Use a python context which does rollback
* Fix pep8
* Full test coverage of module

yvaucher committed:
Add bug tracker link on README.rst

Commit:
[ADD] - document_page_procedure: Added assigned fixed views, updated pot and added spanish translations.

Commit:
[FIX+IMP] mgmtsystem_nonconformity: Fixed typo in data + Added updated es translation.

oca-transbot committed:
OCA Transbot updated translations from Transifex

sbidoul committed:
[UPD] prefix versions with 8.0

pedrobaeza committed:
[MIG] Make modules uninstallable

pedrobaeza committed:
[MIG] Make modules uninstallable

pedrobaeza committed:
[MIG] Make modules uninstallable

naousse committed:
Migration of module mgmtsystem_manual from Odoo 8.0 to Odoo 9.0

naousse committed:
Migration of document_page_procedure to Odoo 9.0

naousse committed:
Migration of document_page_quality_manual to Odoo 9.0

[IMP] Update the structure to ISO 9001:2015

Commit:
[IMP] Add menu Categories

Update README.rst

[IMP] Add menu Categories

[IMP] Menu sequence

[FIX] Manuals action domain

oca-transbot committed:
OCA Transbot updated translations from Transifex

Commit:
Update README.rst

naousse committed:
[IMP+FIX] mgmtsystem_nonconformity: Improvements

* Improving test coverage
* Fixing flake 8 error
* Fixing test error
* Remove unnecessary code

Commit:
[IMP] README.rst

naousse committed:
[FIX+IMP] mgmtsystem_nonconformity: Several patches

* adding drag and drop to kanban view
* Fixing flake 8 error
* Fixing tests errors
* add mgmtsystem_nonconformity_state access rule
* removing unused code
* Improving form and test

dreispt committed:
[FIX+IMP] mgmtsystem_nonconformity: Several things

* Remove forced workflow logic
* Add kanban state
* Fix report state field
* Nonconformity view must load state data
* Adjust tests and make them pass
* Add kanban state (can be used for approvals)
* Fix duplicate stage/state tracking
* Fix report menu
* Add validation on close, requiring evaluation comment
* Replace age by days since last update
* Fix reports
* Reset Kanban State when changing Stages
* Prefer using the standard user_id field
* Properly implement the NC closing date
* Require action plan review for In Progress; Open Actions when In Progress
* Rename Pending Approval to Action Plan

naousse committed:
Removing pylint and flake8 error using comment

import translator _

dreispt committed:
Don't close an NC with open Actions + Add tests to group by stage + Add tests for Kanban state reset on stage change

sbidoul committed:
[FIX] remove en.po that was erroneously created by transbot

sbidoul committed:
[FIX] remove en.po that was erroneously created by transbot

oca-transbot committed:
OCA Transbot updated translations from Transifex

pedrobaeza committed:
[MIG] Make modules uninstallable

pedrobaeza committed:
[MIG] Make modules uninstallable

pedrobaeza committed:
[MIG] Make modules uninstallable

pedrobaeza committed:
[MIG] Rename manifest files

pedrobaeza committed:
[MIG] Rename manifest files

pedrobaeza committed:
[MIG] Rename manifest files

eugen-don committed:
MIG-mgmtsystem_manuals-to-V.-10.0

MIG-mgmtsystem_manuals-to-V.-10.0

eugen-don committed:
MIG-mgmtsystem_nonconformity-to-V.-10.0

MIG-mgmtsystem_nonconformity-to-V.-10.0

eugen-don committed:
MIG document_page_procedure to V. 10.0

MIG document_page_procedure to V. 10.0

eugen-don committed:
MIG document_page_quality_manual to V. 10.0

MIG document_page_quality_manual to V. 10.0

ThomasBinsfeld committed:
[IMP] mgmtsystem: document_page_approval as option

oca-transbot committed:
OCA Transbot updated translations from Transifex

benwillig committed:
[FIX+IMP] mgmsystem_nonconformity: 4 things:

* [CHG] Moved immediate action field in description tab
* [ADD] 'name' attribute on description page to allow inheritance
* [FIX] Added specific tree view for causes with create=1 and delete=1 to allow a simple management user to select/deselect causes
* [FIX] Set create_date as readonly because it can't be changed anyway

jalzaga committed:
[MIG] document_page_procedure: Migration to 11.0

jalzaga committed:
[MIG] mgmtsystem_manual: Migration to 11.0

jalzaga committed:
[MIG] document_page_quality_manual: Migration to 11.0

jalzaga committed:
[MIG] mgmtsystem_nonconformity: Migration to 11.0

oca-transbot committed:
Update translation files

Updated by Update PO files to match POT (msgmerge) hook in Weblate.

oca-transbot committed:
Update translation files

Updated by Update PO files to match POT (msgmerge) hook in Weblate.

primes2h committed:
Added translation using Weblate (Italian)

primes2h committed:
Added translation using Weblate (Italian)

Translated using Weblate (Italian)

Currently translated at 100.0% (4 of 4 strings)

Translation: management-system-11.0/management-system-11.0-mgmtsystem_manual
Translate-URL: https://translation.odoo-community.org/projects/management-system-11-0/management-system-11-0-mgmtsystem_manual/it/

primes2h committed:
Translated using Weblate (Italian)

Currently translated at 95.0% (114 of 120 strings)

Translation: management-system-11.0/management-system-11.0-mgmtsystem_nonconformity
Translate-URL: https://translation.odoo-community.org/projects/management-system-11-0/management-system-11-0-mgmtsystem_nonconformity/it/

gurneyalex committed:
[MIG] migration to 12.0

gurneyalex committed:
[MIG] mgmgsytem_nonconformity to 12.0

Commit:
[MIG] document_page_procedure: migrate to v12

Commit:
[MIG] document_page_quality_manual: migrate to v12

sbidoul committed:
[FIX] bad version number on 12.0 branch

ernestotejeda committed:
[FIX] Fixing pylint errors

Modules: document_page_quality_manual, document_page_work_instruction,
mgmtsystem_action

nicomacr committed:
[PORT] From 11.0

[ADD] mgmtsystem_nonconformity: Add activities to nonconformity

Add the activities to the model "nonconformity"

OCA-git-bot committed:
mgmtsystem_nonconformity 12.0.1.1.0

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (120 of 120 strings)

Translation: management-system-12.0/management-system-12.0-mgmtsystem_nonconformity
Translate-URL: https://translation.odoo-community.org/projects/management-system-12-0/management-system-12-0-mgmtsystem_nonconformity/pt_BR/

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (5 of 5 strings)

Translation: management-system-12.0/management-system-12.0-mgmtsystem_manual
Translate-URL: https://translation.odoo-community.org/projects/management-system-12-0/management-system-12-0-mgmtsystem_manual/pt_BR/

pedrobaeza committed:
[FIX] mgmtsystem_nonconformity: Proper script

Without decorator, you receive a cursor, not an environment

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: management-system-12.0/management-system-12.0-mgmtsystem_nonconformity
Translate-URL: https://translation.odoo-community.org/projects/management-system-12-0/management-system-12-0-mgmtsystem_nonconformity/

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (147 of 147 strings)

Translation: management-system-12.0/management-system-12.0-mgmtsystem_nonconformity
Translate-URL: https://translation.odoo-community.org/projects/management-system-12-0/management-system-12-0-mgmtsystem_nonconformity/pt_BR/

awesome-manuel committed:
Translated using Weblate (German)

Currently translated at 100.0% (5 of 5 strings)

Translation: management-system-12.0/management-system-12.0-mgmtsystem_manual
Translate-URL: https://translation.odoo-community.org/projects/management-system-12-0/management-system-12-0-mgmtsystem_manual/de/

ernestotejeda committed:
[IMP] mgmtsystem_manual: black, isort

ernestotejeda committed:
[MIG] mgmtsystem_manual: Migration to 13.0

ernestotejeda committed:
[IMP] document_page_procedure: black, isort

ernestotejeda committed:
[MIG] document_page_procedure: Migration to 13.0

ernestotejeda committed:
[IMP] document_page_quality_manual: black, isort

ernestotejeda committed:
[MIG] document_page_quality_manual: Migration to 13.0

ernestotejeda committed:
[IMP] mgmtsystem_nonconformity: black, isort

ernestotejeda committed:
[MIG] mgmtsystem_nonconformity: Migration to 13.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: management-system-13.0/management-system-13.0-mgmtsystem_nonconformity
Translate-URL: https://translation.odoo-community.org/projects/management-system-13-0/management-system-13-0-mgmtsystem_nonconformity/

Commit:
Applying multicompany style

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (147 of 147 strings)

Translation: management-system-13.0/management-system-13.0-mgmtsystem_nonconformity
Translate-URL: https://translation.odoo-community.org/projects/management-system-13-0/management-system-13-0-mgmtsystem_nonconformity/pt_BR/

Commit:
FIX mgmtsystem_nonconformity: Default value for company_id

sergiocorato committed:
[FIX] missing compute

Rad0van committed:
Changes to common display of forms

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: management-system-13.0/management-system-13.0-mgmtsystem_nonconformity
Translate-URL: https://translation.odoo-community.org/projects/management-system-13-0/management-system-13-0-mgmtsystem_nonconformity/

flachica committed:
mgmtsystem_nonconformity: Non conformity statusbar not clickable

The Non conformity form's statusbar is intended to be clickable, but Odoo 13 is apparently no longer compatible with clickable="True". Instead, either clickable="1" or options="{'clickable': 1}" must be used. Odoo core code uses the latter, so the Claim form has been updated to match core convention.

flachica committed:
Set name in all pages

In order to facilitate inherits, to include name attribute in all pages

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (149 of 149 strings)

Translation: management-system-13.0/management-system-13.0-mgmtsystem_nonconformity
Translate-URL: https://translation.odoo-community.org/projects/management-system-13-0/management-system-13-0-mgmtsystem_nonconformity/pt_BR/

Commit:
[IMP] mgmtsystem_nonconformity: active field to origins

dzungtran89 committed:
[IMP] document_page_quality_manual: black, isort, prettier

dzungtran89 committed:
[MIG] document_page_quality_manual: Migration to 14.0

dzungtran89 committed:
[IMP] document_page_procedure: black, isort, prettier

dzungtran89 committed:
[MIG] document_page_procedure: Migration to 14.0

dzungtran89 committed:
[IMP] mgmtsystem_manual: black, isort, prettier

dzungtran89 committed:
[MIG] mgmtsystem_manual: Migration to 14.0

dzungtran89 committed:
[IMP] mgmtsystem_nonconformity: black, isort, prettier

dzungtran89 committed:
[MIG] mgmtsystem_nonconformity: Migration to 14.0

victoralmau committed:
[FIX] mgmtsystem_manual: Set correct domain and context in action manual (auto-fill manual category).
TT32895

OCA-git-bot committed:
mgmtsystem_manual 14.0.1.1.0

albariera committed:
[IMP] mgmtsystem_nonconformity: enabling to create non-conformities from different models

victoralmau committed:
[IMP] mgmtsystem_nonconformity: Add button_box div to nonconformity form view to improve inheritance.
TT33874

YvanDotet committed:
[MIG] document_page_quality_manual: Migration to 15.0

[UPD] noupdate data

yankinmax committed:
[15.0][MIG] mgmtsystem_nonconformity

yankinmax committed:
[15.0][FIX] linter issues

YvanDotet committed:
[MIG] document_page_procedure: Migration to 15.0

YvanDotet committed:
[MIG] mgmtsystem_manuel: Migration to 15.0

foah committed:
nonconformity_stage : state field changed to non ReadOnly

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (5 of 5 strings)

Translation: management-system-15.0/management-system-15.0-mgmtsystem_manual
Translate-URL: https://translation.odoo-community.org/projects/management-system-15-0/management-system-15-0-mgmtsystem_manual/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (160 of 160 strings)

Translation: management-system-15.0/management-system-15.0-mgmtsystem_nonconformity
Translate-URL: https://translation.odoo-community.org/projects/management-system-15-0/management-system-15-0-mgmtsystem_nonconformity/it/

FernandoRomera committed:
[16.0][MIG] mgmtsystem_manual: Migration to 16.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: management-system-16.0/management-system-16.0-mgmtsystem_manual
Translate-URL: https://translation.odoo-community.org/projects/management-system-16-0/management-system-16-0-mgmtsystem_manual/

FernandoRomera committed:
[16.0][MIG] document_page_procedure: Migration to 16.0

FernandoRomera committed:
[16.0][MIG] document_page_quality_manual: Migration to 16.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: management-system-16.0/management-system-16.0-document_page_quality_manual
Translate-URL: https://translation.odoo-community.org/projects/management-system-16-0/management-system-16-0-document_page_quality_manual/

FernandoRomera committed:
[16.0][MIG] mgmtsystem_nonconformity: Migration to 16.0

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (157 of 157 strings)

Translation: management-system-16.0/management-system-16.0-mgmtsystem_nonconformity
Translate-URL: https://translation.odoo-community.org/projects/management-system-16-0/management-system-16-0-mgmtsystem_nonconformity/it/

sysadminmatmoz committed:
Translated using Weblate (Slovenian)

Currently translated at 58.2% (92 of 158 strings)

Translation: management-system-16.0/management-system-16.0-mgmtsystem_nonconformity
Translate-URL: https://translation.odoo-community.org/projects/management-system-16-0/management-system-16-0-mgmtsystem_nonconformity/sl/

etobella committed:
[FIX] mgmtsystem_nonconformity:  Use child_of

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (158 of 158 strings)

Translation: management-system-16.0/management-system-16.0-mgmtsystem_nonconformity
Translate-URL: https://translation.odoo-community.org/projects/management-system-16-0/management-system-16-0-mgmtsystem_nonconformity/es/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (5 of 5 strings)

Translation: management-system-16.0/management-system-16.0-mgmtsystem_manual
Translate-URL: https://translation.odoo-community.org/projects/management-system-16-0/management-system-16-0-mgmtsystem_manual/es/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 99.3% (157 of 158 strings)

Translation: management-system-16.0/management-system-16.0-mgmtsystem_nonconformity
Translate-URL: https://translation.odoo-community.org/projects/management-system-16-0/management-system-16-0-mgmtsystem_nonconformity/es/

yassine-rhimi committed:
Update mgmtsystem_nonconformity_origin.py

add unaccent=False

yassine-rhimi committed:
Update mgmtsystem_nonconformity_cause.py

    parent_path = fields.Char(index=True, unaccent=False)

yassine-rhimi committed:
Update mgmtsystem_nonconformity.py

yassine-rhimi committed:
Update mgmtsystem_nonconformity.py

grindtildeath committed:
Add missing noupdate flag on nonconformity stages

grindtildeath committed:
Add migration script to set noupdate flag

m1k3lm committed:
[IMP] document_page_procedure: pre-commit auto fixes

m1k3lm committed:
[MIG] document_page_procedure: Migration to 17.0

etobella committed:
[IMP] mgmtsystem_nonconformity: Show non conformities on all kind of views

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (159 of 159 strings)

Translation: management-system-16.0/management-system-16.0-mgmtsystem_nonconformity
Translate-URL: https://translation.odoo-community.org/projects/management-system-16-0/management-system-16-0-mgmtsystem_nonconformity/it/

Commit:
[IMP] mgmtsystem_manual: pre-commit auto fixes

Commit:
[MIG] mgmtsystem_manual: Migration to 17.0

Borruso committed:
[FIX]  mgmtsystem_nonconformity: fix expected singleton

mostafabarmshory committed:
Added translation using Weblate (Persian)

mostafabarmshory committed:
Added translation using Weblate (Persian)

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 89.9% (143 of 159 strings)

Translation: management-system-16.0/management-system-16.0-mgmtsystem_nonconformity
Translate-URL: https://translation.odoo-community.org/projects/management-system-16-0/management-system-16-0-mgmtsystem_nonconformity/pt_BR/

xesodoo committed:
[16.0][FIX] mgmtsystem_nonconformity: Fix origin and cause form views

The origin and cause form views do not render correctly because the description
field in both forms uses colspan="4". Removing the colspan attribute so that
the default value is used fixes the problem.

jakobkrabbe committed:
Translated using Weblate (Swedish)

Currently translated at 100.0% (159 of 159 strings)

Translation: management-system-16.0/management-system-16.0-mgmtsystem_nonconformity
Translate-URL: https://translation.odoo-community.org/projects/management-system-16-0/management-system-16-0-mgmtsystem_nonconformity/sv/

mostafabarmshory committed:
Added translation using Weblate (Persian)

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: management-system-17.0/management-system-17.0-mgmtsystem_manual
Translate-URL: https://translation.odoo-community.org/projects/management-system-17-0/management-system-17-0-mgmtsystem_manual/

EdgarRetes committed:
[MIG] mgmtsystem_nonconformity: Migration to 17.0

[FIX] mgmtsystem_nonconformity: pre-commit

EdgarRetes committed:
[FIX] mgmtsystem_nonconformity: fields attributes

EdgarRetes committed:
[MIG] mgmtsystem_nonconformity: Migration to 17.0

natuan9 committed:
[IMP] mgmtsystem_nonconformity: pre-commit auto fixes

natuan9 committed:
[MIG] mgmtsystem_nonconformity: Migration to 18.0

natuan9 committed:
[IMP] document_page_procedure: pre-commit auto fixes

natuan9 committed:
[MIG] document_page_procedure: Migration to 18.0

natuan9 committed:
[IMP] mgmtsystem_manual: pre-commit auto fixes

natuan9 committed:
[MIG] mgmtsystem_manual: Migration to 18.0

Kimkhoi3010 committed:
[IMP] document_page_quality_manual: pre-commit auto fixes

Kimkhoi3010 committed:
[MIG] document_page_quality_manual: Migration to 18.0

natuan9 committed:
add checklog-odoo.cfg

thienvh332 commented on mgmtsystem_nonconformity/models/mgmtsystem_nonconformity_abstract.py:
```diff
@@ -0,0 +1,90 @@
+
+
+class MgmtsystemNonconformityAbstract(models.AbstractModel):
+    # TODO: Remove this on 17.0 and move everything on mail.thread
```
Get it done bro :wink:

thienvh332 commented on mgmtsystem_nonconformity/views/mgmtsystem_nonconformity.xml:
```diff
@@ -0,0 +1,397 @@
+        <field name="name">mgmtsystem.nonconformity.kanban</field>
+        <field name="model">mgmtsystem.nonconformity</field>
+        <field name="arch" type="xml">
+            <kanban
```
Edit the kanban view
![image](https://github.com/user-attachments/assets/3bd9731c-8688-4c0c-a366-95d174088ffa)

thienvh332 commented on mgmtsystem_nonconformity/views/mgmtsystem_nonconformity.xml:
```diff
@@ -0,0 +1,397 @@
+                                        </li>
+                                    </t>
+                                </ul>
+                            </div>
```
Do we still need it?

thienvh332 commented on mgmtsystem_nonconformity/models/mgmtsystem_nonconformity_abstract.py:
```diff
@@ -0,0 +1,90 @@
+        "mgmtsystem.nonconformity",
+        inverse_name="res_id",
+        domain=lambda r: [("res_model", "=", r._name)],
+        readonly=True,
```
If a field is marked as readonly=True on the model, it won't be possible to import it through the Odoo import tool, so avoid it as possible and define the readonly attribute on the views instead.
```suggestion
```

thienvh332 commented on mgmtsystem_nonconformity/models/mgmtsystem_nonconformity.py:
```diff
@@ -0,0 +1,247 @@
+        store=True,
+        readonly=True,
+    )
+    closing_date = fields.Datetime(readonly=True)
```
Same here

thienvh332 commented on mgmtsystem_nonconformity/models/mgmtsystem_nonconformity.py:
```diff
@@ -0,0 +1,247 @@
+        "# of days to close",
+        compute="_compute_number_of_days_to_close",
+        store=True,
+        readonly=True,
```
Same here. Let's fix it all :wink:

natuan9 commented:
Hi a @thienvh332 , Please help to review PR again

**Kanban Migration**
- Version 17.0
![image](https://github.com/user-attachments/assets/944f2a80-e366-4032-bead-110d0679e073)

- Version 18.0
![image](https://github.com/user-attachments/assets/996fe9bb-2821-47be-b568-22bfa15599a2)

thienvh332 commented on mgmtsystem_nonconformity/models/mgmtsystem_nonconformity.py:
```diff
@@ -19,27 +19,24 @@ def _default_stage(self):
         ].search([("is_starting", "=", True)], limit=1)
 
     @api.model
-    def _stage_groups(self, stages, domain, order):
```
Is there any reason for this removal?

natuan9 replied:
order parameter of group_expand methods is remove
you can see the detail here `https://github.com/odoo/odoo/pull/139294/commits/6e6bcbc6ab5cef679b17b617d821b2bb3f778af8`

thienvh332 commented on mgmtsystem_nonconformity/reports/mgmtsystem_nonconformity_report.xml:
```diff
@@ -0,0 +1,47 @@
+            <pivot string="Non conformity" disable_linking="True">
+                <field name="state" type="row" />
+                <field name="user_id" type="col" />
+                <field name="number_of_nonconformities" type="measure" />
```
```suggestion
                <field name="number_of_nonconformities" type="measure" readonly="True"  />
```

thienvh332 commented on mgmtsystem_nonconformity/reports/mgmtsystem_nonconformity_report.xml:
```diff
@@ -0,0 +1,47 @@
+            <graph string="Non conformity">
+                <field name="state" />
+                <field name="user_id" />
+                <field name="number_of_nonconformities" type="measure" />
```
```suggestion
                <field name="number_of_nonconformities" type="measure" readonly="True"  />
```

thienvh332 commented on mgmtsystem_nonconformity/views/mgmtsystem_nonconformity.xml:
```diff
@@ -8,11 +8,11 @@
-            <tree>
-                <field name="ref" />
+            <list>
+                <field name="ref" readonly="1" />
```
```suggestion
                <field name="ref" readonly="True" />
```

thienvh332 commented on mgmtsystem_nonconformity/models/mgmtsystem_nonconformity_cause.py:
```diff
@@ -0,0 +1,34 @@
+    name = fields.Char("Cause", required=True, translate=True)
+    description = fields.Text()
+    sequence = fields.Integer(help="Defines the order to present items")
+    parent_path = fields.Char(index=True, unaccent=False)
```
Remove **`unaccent`** in all fields. It is not necessary anymore.

thienvh332 commented on mgmtsystem_nonconformity/views/mgmtsystem_origin.xml:
```diff
@@ -0,0 +1,76 @@
+        <field name="arch" type="xml">
+            <form string="Origins of nonconformity">
+                <sheet>
+                    <field name="active" invisible="1" />
```
```suggestion

```

thienvh332 commented on mgmtsystem_nonconformity/views/mgmtsystem_nonconformity.xml:
```diff
@@ -0,0 +1,357 @@
+                        widget="statusbar"
+                        options="{'clickable': 1}"
+                    />
+                    <field name="state" invisible="True" />
```
```suggestion

```

thienvh332 commented on mgmtsystem_nonconformity/models/mgmtsystem_nonconformity_origin.py:
```diff
@@ -0,0 +1,33 @@
+    name = fields.Char("Origin", required=True, translate=True)
+    description = fields.Text()
+    sequence = fields.Integer(help="Defines the order to present items")
+    parent_path = fields.Char(index=True, unaccent=False)
```
Remove here

thienvh332 reviewed (approved):
LGTM

OCA-git-bot committed:
Merge PR #610 into 18.0

Signed-off-by max3903

Commit:
[UPD] Update document_page_procedure.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: management-system-18.0/management-system-18.0-document_page_procedure
Translate-URL: https://translation.odoo-community.org/projects/management-system-18-0/management-system-18-0-document_page_procedure/

OCA-git-bot committed:
Merge PR #606 into 18.0

Signed-off-by max3903

Commit:
[UPD] Update mgmtsystem_manual.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #608 into 18.0

Signed-off-by max3903

Commit:
[UPD] Update document_page_quality_manual.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: management-system-18.0/management-system-18.0-document_page_quality_manual
Translate-URL: https://translation.odoo-community.org/projects/management-system-18-0/management-system-18-0-document_page_quality_manual/
